src/util/intOps.ml

```diff
@@ -192,10 +192,14 @@ struct
   let mul = Big_int_Z.mult_big_int
 
   (* If the first operand of a div is negative, Zarith rounds the result away from zero.
-    We thus always transform this into a divison with a non-negative first operand.
+     We thus always transform this into a division with a non-negative first operand.
   *)
   let div a b = if Big_int_Z.lt_big_int a zero then Big_int_Z.minus_big_int (Big_int_Z.div_big_int (Big_int_Z.minus_big_int a) b) else Big_int_Z.div_big_int a b
-  let rem = Big_int_Z.mod_big_int
+
+  (* Big_int_Z.mod_big_int computes the Euclidian Modulus, but what we want here is the remainder, as returned by mod on ints
+     -1 rem 5 == -1, whereas -1 Euclid-Mod 5 == 4
+  *)
```

Comment:
I assume this is already the case with other `IntOps` implementations, right? Maybe this should also be stated for `val rem` in its signature documentation to make it clear that we want this behavior consistently for all of them.

Reply:
I added a comment. Yes, all other implementations compute the remainder.